mabe.go

```diff
@@ -0,0 +1,256 @@
+)
+
+// Mabe schema constants
+const (
+	MabeVersion        = "1.0"
+	MabeNamespace      = "https://recepcionfe.mabempresa.com/cfd/addenda/v1"
+	MabeSchemaLocation = "https://recepcionfe.mabempresa.com/cfd/addenda/v1/mabev1.xsd"
+	MabeNotApplicable  = "NA"
+)
+
+// TipoDocumento valid values
+const (
+	MabeTipoDocumentoFactura     = "FACTURA"
+	MabeTipoDocumentoNotaCredito = "NOTA CREDITO"
+	MabeTipoDocumentoNotaCargo   = "NOTA CARGO"
+)
+
+// MabeFactura is the root element of the Mabe addendum
+type MabeFactura struct {
+	XMLName        xml.Name `xml:"mabe:Factura"`
+	Namespace      string   `xml:"xmlns:mabe,attr"`
+	SchemaLocation string   `xml:"xsi:schemaLocation,attr"`
+
+	Version       string `xml:"version,attr"`
+	TipoDocumento string `xml:"tipoDocumento,attr"`
+	Folio         string `xml:"folio,attr"`
+	Fecha         string `xml:"fecha,attr"`
+	OrdenCompra   string `xml:"ordenCompra,attr"`
+	Referencia1   string `xml:"referencia1,attr"`
+	Referencia2   string `xml:"referencia2,attr,omitempty"`
+
+	Moneda    *MabeMoneda     `xml:"mabe:Moneda"`
+	Proveedor *MabeProveedor  `xml:"mabe:Proveedor"`
+	Entrega   *MabeEntrega    `xml:"mabe:Entrega"`
+	Detalles  *[]*MabeDetalle `xml:"mabe:Detalles>mabe:Detalle"`
+
+	Descuentos  *MabeDescuentos  `xml:"mabe:Descuentos,omitempty"`
+	Subtotal    *MabeImporte     `xml:"mabe:Subtotal"`
+	Traslados   *[]*MabeImpuesto `xml:"mabe:Traslados>mabe:Traslado"`
+	Retenciones *[]*MabeImpuesto `xml:"mabe:Retenciones>mabe:Retencion"`
+	Total       *MabeImporte     `xml:"mabe:Total"`
+}
+
+// MabeMoneda carries the data about the invoice's currency
+type MabeMoneda struct {
+	TipoMoneda      string `xml:"tipoMoneda,attr"`
+	TipoCambio      string `xml:"tipoCambio,attr,omitempty"`      // Not implemented yet
+	ImporteConLetra string `xml:"importeConLetra,attr,omitempty"` // Not implemented yet
+}
+
+// MabeProveedor carries the data about the invoice's supplier
+type MabeProveedor struct {
+	Codigo string `xml:"codigo,attr"`
+}
+
+// MabeEntrega carries the data about the invoice's delivery
+type MabeEntrega struct {
+	PlantaEntrega string `xml:"plantaEntrega,attr"`
+	Calle         string `xml:"calle,attr,omitempty"`
+	NoExterior    string `xml:"noExterior,attr,omitempty"`
+	NoInterior    string `xml:"noInterior,attr,omitempty"`
+	CodigoPostal  string `xml:"codigoPostal,attr,omitempty"`
+}
+
+// MabeDetalle carries the data about one invoice's line
+type MabeDetalle struct {
+	NoLineaArticulo int    `xml:"noLineaArticulo,attr"`
+	CodigoArticulo  string `xml:"codigoArticulo,attr"`
+	Descripcion     string `xml:"descripcion,attr"` //nolint:misspell
+	Unidad          string `xml:"unidad,attr"`
+	Cantidad        string `xml:"cantidad,attr"`
+	PrecioSinIva    string `xml:"precioSinIva,attr"`
+	ImporteSinIva   string `xml:"importeSinIva,attr"`
+	PrecioConIva    string `xml:"precioConIva,attr,omitempty"`  // Not implemented yet
+	ImporteConIva   string `xml:"importeConIva,attr,omitempty"` // Not implemented yet
```

Comment:
I left these two fields out of scope because they're optional and I don't think they add any value